guidance/library/_json.py

```diff
@@ -329,9 +326,14 @@ def _gen_json_any(lm):
 @guidance(stateless=True)
 def _gen_json(
     lm,
-    json_schema: Mapping[str, Any],
+    json_schema: JSONSchema,
     definitions: Mapping[str, Callable[[], GrammarFunction]],
 ):
+    if json_schema is True:
+        json_schema = {}
+    elif json_schema is False:
+        raise ValueError("No valid JSON can be generated from a schema of `False`")
```

Comment:
thoughts on raising error vs. returning empty string?

Reply:
Intuitively I feel like warning + empty string is more "correct", but I'm easily convinced otherwise.

Reply:
The empty string isn't valid against the `False` schema. I think we need to raise an exception if someone wants to generate against a schema with no valid instances. Another "bad" schema by this standard is
```
{
  "type": "object",
  "properties": {
    "name": false
  },
  "required": ["name"]
  }
  ```
  
 Note that raising an exception upon encountering a `false` subschema isn't actually correct though. See the following example:
 ```
{
  "type": "object",
  "properties": {
    "name": false
  },
  "required": []
  }
```
This should validate against **any object that does not have the key `"name"`**. Will not support for now -- I think I'll open an issue against this just to track it...